src/Domain/FunctionsOfTime/FunctionOfTime.hpp

```diff
@@ -55,6 +55,12 @@ class FunctionOfTime : public PUP::able {
   /// interval.
   virtual std::array<double, 2> time_bounds() const = 0;
 
+  /// The smallest value `time_bounds()[1]` could have that is greater
+  /// than \p time.  For updatable functions of time, this will be the
+  /// expiration time of the last update before or at \p time.  For
+  /// non-updatable functions, this will be infinity.
+  virtual double expiration_after(double time) const = 0;
```

Comment:
Maybe don't make this pure virtual. Make the default return infinity like how we have the `update` function error by default. Then you don't have to make changes to `FixedSpeedCubic` or `SettleToConstant`.